File: Day 2/day2-part1.js
const fetchInput = require("./fetchInput");

console.log("Day 2 started!");

function getAllInvalidIds(input) {
  const invalidIds = [];
  const ranges = input.trim().split(",");
  for (const range of ranges) {
    const [id1, id2] = range.split("-");
    let num1 = Number(id1);
    let num2 = Number(id2);
    while (num1 <= num2) {
      const numStr = num1.toString();
      if (numStr.length % 2 === 0) {
        const mid = numStr.length / 2;
        const part1 = numStr.slice(0, mid);
        const part2 = numStr.slice(mid);
        if (part1 === part2) {
          invalidIds.push(num1);
        }
      }
      num1++;
    }
  }
  return invalidIds;
}

function sumInvalidIds(invalidIds) {
  let total = 0;
  for (let i = 0; i < invalidIds.length; i++) {
    total += invalidIds[i];
  }
  return total;
}

async function main() {
  try {
    const rawInput = await fetchInput();

    console.log(sumInvalidIds(getAllInvalidIds(rawInput)));
  } catch (error) {
    console.error("Error fetching or processing input:", error);
  }
}

main();

File: Day 2/cSharp/day2-part1.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;

#if PART1
class Program
{
    static void Main(string[] args)
    {
        Console.WriteLine("Day 2 Part 1 started!");
        try
        {
            string input = File.ReadAllText("input-day2.md").Trim();
            string[] ranges = input.Split(',');
            var invalidIds = new List<long>();

            foreach (var range in ranges)
            {
                string[] parts = range.Split('-');
                long start = long.Parse(parts[0]);
                long end = long.Parse(parts[1]);
                for (long i = start; i <= end; i++)
                {
                    string numStr = i.ToString();
                    if (numStr.Length % 2 == 0)
                    {
                        int mid = numStr.Length / 2;
                        string part1 = numStr.Substring(0, mid);
                        string part2 = numStr.Substring(mid);
                        if (part1 == part2)
                        {
                            invalidIds.Add(i);
                        }
                    }
                }
            }
            Console.WriteLine(invalidIds.Sum());
        }
        catch (Exception ex)
        {
            Console.Error.WriteLine($"Error reading or processing input: {ex.Message}");
        }
    }
}
#endif